ui/src/time_picker.tsx

```diff
@@ -0,0 +1,259 @@
+// Copyright 2020 H2O.ai, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react'
+import * as Fluent from '@fluentui/react'
+import { B, Id, S, U } from 'h2o-wave'
+import { cssVar } from './theme'
+import { wave } from './ui'
+import { stylesheet } from 'typestyle'
+import { PopperProps, TextFieldProps, Theme, ThemeOptions } from '@mui/material'
+import DateFnsUtils from '@date-io/date-fns'
+import { VirtualElement } from '@popperjs/core/lib'
+import { CalendarOrClockPickerView } from '@mui/x-date-pickers/internals/models'
+
+/**
+ * Create a time picker.
+ *
+ * A time picker allows a user to pick a time value.
+ */
+
+export interface TimePicker {
+  /** An identifying name for this component. */
+  name: Id
+  /** Text to be displayed alongside the component. */
+  label?: S
+  /** A string that provides a brief hint to the user as to what kind of information is expected in the field. */
+  placeholder?: S
+  /** The time value in hh:mm format. E.g. '10:30', '14:25', '23:59', '00:00' */
+  value?: S
+  /** True if this field is disabled. */
+  disabled?: B
+  /** The width of the time picker, e.g. '100px'. Defaults to '100%'. */
+  width?: S
+  /** True if the component should be visible. Defaults to True. */
+  visible?: B
+  /** True if the form should be submitted when the time is selected. */
+  trigger?: B
+  /** True if this is a required field. Defaults to False. */
+  required?: B
+  /** Specifies 12-hour or 24-hour time format. One of `12` or `24`. Defaults to `24`. */
+  hour_format?: S
```

Comment:
I guess default here should be `12`?
```suggestion
  /** Specifies 12-hour or 24-hour time format. One of `12` or `24`. Defaults to `12`. */
  hour_format?: S
```

Reply:
My fault, thanks for noticing!